patch-package
--- a/node_modules/karma-safari-launcher/index.js
+++ b/node_modules/karma-safari-launcher/index.js
@@ -11,7 +11,9 @@ var SafariBrowser = function(baseBrowserDecorator) {
 
     fs.readFile(HTML_TPL, function(err, data) {
       var content = data.toString().replace('%URL%', url);
-      var staticHtmlPath = self._tempDir + '/redirect.html';
+      // var staticHtmlPath = self._tempDir + '/redirect.html';
+      var staticHtmlPath = path.join(process.env.HOME, 'Library/Containers/com.apple.Safari/Data/redirect.html');
+
 
       fs.writeFile(staticHtmlPath, content, function(err) {
         self._execCommand(self._getCommand(), [staticHtmlPath]);
